modules/loader/src/main/java/com/opengamma/strata/loader/csv/IborCapFloorTradeCsvPlugin.java

```diff
@@ -0,0 +1,330 @@
+      .build();
+
+  //-------------------------------------------------------------------------
+  @Override
+  public Set<String> tradeTypeNames() {
+    return ImmutableSet.of("CAPFLOOR");
+  }
+
+  @Override
+  public Optional<Trade> parseTrade(
+      Class<?> requiredJavaType,
+      CsvRow baseRow,
+      List<CsvRow> additionalRows,
+      TradeInfo info,
+      TradeCsvInfoResolver resolver) {
+
+    if (requiredJavaType.isAssignableFrom(IborCapFloorTrade.class)) {
+      return Optional.of(resolver.parseCapFloorTrade(baseRow, info));
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public String getName() {
+    return "CapFloor";
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Parses from the CSV row.
+   *
+   * @param row the CSV row
+   * @param info the trade info
+   * @param resolver the resolver used to parse additional information
+   * @return the parsed trade
+   */
+  static IborCapFloorTrade parseCapFloor(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    IborCapFloorTrade iborCapFloorTrade = parseRow(row, info);
+    return resolver.completeTrade(row, iborCapFloorTrade);
+  }
+
+  //-------------------------------------------------------------------------
+  // parses the row to a trade
+  private static IborCapFloorTrade parseRow(CsvRow row, TradeInfo info) {
+    IborCapFloor iborCapFloor = parseIborCapFloor(row);
+    Optional<AdjustablePayment> paymentOpt = parsePremium(row);
+    return paymentOpt.map(adjustablePayment ->
+        IborCapFloorTrade.builder()
+            .info(info)
+            .product(iborCapFloor)
+            .premium(adjustablePayment)
+            .build())
+        .orElseGet(() -> IborCapFloorTrade.builder()
+            .info(info)
+            .product(iborCapFloor)
+            .build());
+  }
+
+  // parse an IborCapFloor
+  private static IborCapFloor parseIborCapFloor(CsvRow row) {
+    PayReceive payReceive = row.getValue(DIRECTION_FIELD, LoaderUtils::parsePayReceive);
+    Currency currency = row.getValue(CURRENCY_FIELD, LoaderUtils::parseCurrency);
+    ValueSchedule strike = ValueSchedule.of(row.getValue(STRIKE_FIELD, LoaderUtils::parseDoublePercent));
+    double notional = row.getValue(NOTIONAL_FIELD, LoaderUtils::parseDouble);
+    IborIndex iborIndex = parseIborIndex(row);
+    PeriodicSchedule paymentSchedule = parseSchedule(row, currency);
+    IborCapFloorLeg.Builder capFloorLegBuilder = IborCapFloorLeg.builder()
+        .payReceive(payReceive)
+        .paymentSchedule(paymentSchedule)
+        .currency(currency)
+        .notional(ValueSchedule.of(notional))
+        .calculation(IborRateCalculation.of(iborIndex));
+    String capFloorType = row.getValue(CAP_FLOOR_FIELD);
+    if (capFloorType.toUpperCase(Locale.ENGLISH).equals("CAP")) {
+      capFloorLegBuilder.capSchedule(strike);
+    } else if (capFloorType.toUpperCase(Locale.ENGLISH).equals("FLOOR")) {
```

Comment:
Looking at the API, I would assume that a leg can have both a cap and a floor. There does seem to be validation against it though. Might need to investigate changing things to a single schedule and a type in another PR.

Reply:
A cap/floor would always only have one or the other; the payoff is (Rate fixing - Strike) or (Strike - Rate Fixing)

Reply:
The `IborCapFloorLeg` only lets you have one, but I think the point is that it has a cap schedule and floor schedule as optional properties (even though there is validation that won't allow it). It might be better having a `CapFloor` property and then just a single schedule, and the payoff direction can be determined by the `CapFloor` type, which also aligns with how the parser works/ is represented in CSV format.